internal/metrics/metrics_test.go

```diff
@@ -0,0 +1,76 @@
+package metrics_test
+
+import (
+	"github.com/jakub-dzon/k4e-operator/internal/metrics"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	dto "github.com/prometheus/client_model/go"
+	ctrlmetrics "sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+const (
+	numberOfEdgeDevicesSuccessfulRegisteredValue = 3
+	numberOfEdgeDevicesFailedToRegisterValue     = 1
+	numberOfCreatedOBCsValue                     = 2
+)
+
+func TestMetrics(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Metrics Suite")
+}
+
+func findMetric(src []*dto.MetricFamily, query string) *dto.MetricFamily {
+	for _, s := range src {
+		if s.Name != nil && *s.Name == query {
+			return s
+		}
+	}
+	return nil
+}
+
+func validateMetric(query string, value int){
+	data, err := ctrlmetrics.Registry.Gather()
+	Expect(err).NotTo(HaveOccurred())
+
+	m := findMetric(data, query)
+	Expect(m).ToNot(BeNil(), "metric for %s could not be found", query)
+	Expect(m.Metric).To(HaveLen(1))
+	Expect(m.Metric[0].Counter).ToNot(BeNil())
+	Expect(m.Metric[0].Counter.Value).ToNot(BeNil())
+	Expect(*m.Metric[0].Counter.Value).To(BeEquivalentTo(value))
+}
+var _ = Describe("Metrics", func() {
+	m := metrics.New()
```

Comment:
Just a thing, I will move this to the BeforeEach section. Mainly because you want to have a clear object on Each Test and to avoid future test flakes.

Reply:
Unfortunately metrics object cannot be cleared